ssntp/client.go

```diff
@@ -77,6 +77,7 @@ type Client struct {
 	closed    chan struct{}
 
 	frameWg              sync.WaitGroup
+	frameWgMutex         sync.Mutex
```

Comment:
https://golang.org/pkg/sync/#WaitGroup talks about the "main" goroutine and I've only ever seen examples in that way where a parent creates the waitgroup and awaits children.

In the case of a https://golang.org/pkg/sync/#Cond waiting there is noted to be a lock involved internally, but not with a waitgroup.  Perhaps it is there and opaque, but this go-race logs implies it is not.

This is perhaps a usage where a channel is more appropriate actually instead of a waitgroup.  @sameo what do you think of changing frameWg to a channel?